aten/src/ATen/native/sparse/ValidateCompressedIndicesCommon.h

```diff
@@ -240,6 +240,29 @@ void _validate_compressed_sparse_indices_kernel(
   // For TensorIterator's output: no void lambdas.
   const auto dummy = at::empty({1}, cidx.options());
 
+  // Catch integer overflow from large dimensions. Otherwise, the
+  // invariant checks may fail with bogus exceptions or succeed with
+  // false-positive results when int64_t typed dimensions are cast to
+  // index dtype that corresponds to smaller interger type such as
+  // int32_t.
+  {
+    AT_DISPATCH_INDEX_TYPES(idx.scalar_type(), NAME, [cdim, dim, nnz]() {
+      if (cdim_name == CDimName::CRow) {
```

Comment:
nit: probably would make sense to put that at the very top?

Reply:
yes, we can do that, although, it does not matter much because the tests above use int64 comparisons and there exist no integer overflow issues.